lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -35,6 +35,7 @@ def authorize(money, creditcard, options = {})
         post[:card] = credit_card_hash(creditcard)
         post[:billingAddress] = billing_address_hash(options) if options[:billing_address]
         post[:deliveryAddress] = shipping_address_hash(options) if options[:shipping_address]
+        add_threeds(post, options) if options[:execute_threed]       
```

Comment:
Maybe this is a pattern, but if it isn't, any chance we could make this be either `add_three_ds`/`execute_three_d` or maybe `add_3ds`/`execute_3ds`? `threeds` is a bit hard to read.